consensus/polybft/hash.go

```diff
@@ -0,0 +1,27 @@
+package polybft
+
+import (
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+// polyBFTHeaderHash defines the custom implementation for getting the header hash,
+// because of the extraData field
+func setupHeaderHashFunc() {
+	originalHeaderHash := types.HeaderHash
+
+	types.HeaderHash = func(h *types.Header) types.Hash {
+		// when hashing the block for signing we have to remove from
+		// the extra field the seal and committed seal items
+		extra, err := GetIbftExtraClean(h.ExtraData)
+		if err != nil {
+			// TODO: this should not happen but we have to handle it somehow
+			panic(err)
+		}
```

Comment:
What do you think about returning an error? Panics should not be introduced in the codebase